### PR TITLE
Remove index_realtime and index_realtime_appenderator tasks.

### PR DESCRIPTION
**Pull request by gianm**
`index_realtime` tasks were removed from the documentation in #13107. Even at that time, they weren't really documented per se— just mentioned. They existed solely to support Tranquility, which is an obsolete ingestion method that predates migration of Druid to ASF and is no longer being maintained. Tranquility docs were de-linked from the sidebars and other docs earlier, in #11134. Only a stub remains today, so people with links to the page can see that it's no longer recommended.

`index_realtime_appenderator` tasks existed in the code base, but were never documented, nor as far as I am aware were they used for any purpose.

Since they are both obsolete, this patch removes both task types completely, as well as removes all supporting code that was otherwise unused. It also updates the stub doc for Tranquility to be firmer that it is not compatible. (Previously, the stub doc said it wasn't recommended, and pointed out that it is built against an ancient 0.9.2 version of Druid.)

This patch also has some related metrics changes:

- rename FireDepartmentMetrics to SegmentGenerationMetrics
- removes a useless call to `fireDepartmentMetrics.incrementProcessed()` in `SinglePhaseSubTask` (this metric was not being emitted anyway)

### CODE REVIEW

**Review by clintropolis — APPROVED** [review]

File: server/src/main/java/org/apache/druid/segment/indexing/RealtimeTuningConfig.java [hunk 1]
```diff
@@ -46,8 +42,6 @@ public class RealtimeTuningConfig implements AppenderatorConfig
 {
   private static final Period DEFAULT_INTERMEDIATE_PERSIST_PERIOD = new Period("PT10M");
```
Comment (kfaraz) [inline server/src/main/java/org/apache/druid/segment/indexing/RealtimeTuningConfig.java:43]: I think most of these constants have been moved to `SeekableStreamIndexTaskTuningConfig`. Do we still need this class?

**Review by kfaraz — APPROVED** [review]
LGTM 🚀 !!

File: server/src/main/java/org/apache/druid/segment/indexing/RealtimeTuningConfig.java (hunk 1 above)
Comment (gianm) [reply server/src/main/java/org/apache/druid/segment/indexing/RealtimeTuningConfig.java:43]: Ah, no, we don't need it. I went through all the trouble of adjusting all the call sites to not use this, but then forgot to delete it. Fixed!

**Comment by gianm** [comment]
Hmm. `ITUnionQueryTest` needs to be rewritten somehow. It is making sure `union` datasources work with realtime tasks, and uses an `index_realtime` to do so. It probably needs to be moved to use Kafka.

**Comment by gianm** [comment]
Closed in favor of #16602.

**Pull request closed without merging**